oxenss/rpc/request_handler.cpp

```diff
@@ -1034,6 +1055,84 @@ void RequestHandler::process_client_req(rpc::expire_msgs&& req, std::function<vo
         reply_or_fail(std::move(res));
 }
 
+void RequestHandler::process_client_req(rpc::batch&& req, std::function<void(rpc::Response)> cb) {
+
+    auto subresults = std::make_shared<json>(json::array());
+    for (size_t i = 0; i < req.subreqs.size(); i++)
+        subresults->emplace_back();
+
+    for (size_t i = 0; i < req.subreqs.size(); i++) {
+        auto handler = [subresults, i, cb](Response r) {
+            json& subres = (*subresults)[i];
+            subres["code"] = r.status.first;
+            if (auto* j = std::get_if<json>(&r.body))
+                subres["body"] = std::move(*j);
+            else
+                subres["body"] = std::string{view_body(r)};
+            bool done = true;
+            for (auto& sr : *subresults)
+                if (sr.is_null()) {
+                    done = false;
+                    break;
+                }
+            if (done)
+                cb(Response{http::OK, json({{"results", std::move(*subresults)}})});
+        };
+        std::visit(
+                [this, handler = std::move(handler)](auto&& s) {
+                    process_client_req(std::move(s), std::move(handler));
+                },
+                req.subreqs[i]);
+    }
+}
+
+namespace {
+    struct sequence_manager {
+        std::vector<client_subrequest> subreqs;
+        json subresults = json::array();
+        std::function<void(Response r)> subresult_callback;
+    };
+}  // namespace
+
+void RequestHandler::process_client_req(
+        rpc::sequence&& req, std::function<void(rpc::Response)> cb) {
+
+    if (req.subreqs.empty()) {
+        cb(Response{http::OK, json({{"results", json::array()}})});
```

Comment:
should this be a 400? it seems like a client should not be able to send an empty array and have it be okay with it.

Reply:
I'm kind of indifferent about it; I'll change it.